src/lib/factories.tsx

```diff
@@ -56,6 +57,14 @@ export function createShorthand(
   const valIsPropsObject = _.isPlainObject(value)
   const valIsReactElement = React.isValidElement(value)
 
+  // just return ReactElements that have different type than Component
```

Comment:
@kuzhelov 
```jsx
// and this will be processed DIFFERENTLY from second example!
<Button icon={<MyIcon name='...' bordered='true' circular='true' />} />
```
the whole purpose of this change is not to support examples like this; the arguments are:
- `MyIcon` can have implementation that is very different than `Icon` so the `Icon` component factory (aka `Icon.create`) won't know how to process its props; please refer to the examples in the description for more details or we can have a sync tomorrow;
- this API (passing random elements to known component slots) is almost never used;
- user will still have the ability to achieve this scenario using the `renderIcon` prop **so we're not eliminating existing functionality**
```jsx
// and this will be processed DIFFERENTLY from second example!
<Button renderIcon={(Icon, iconProps) => <MyIcon name='...' bordered='true' circular='true' />} />
```